File: integration-test/java/quarkus-app/src/main/java/de/qualityminds/lazyval/integration/Order.java
package de.qualityminds.lazyval.integration;

import de.qualityminds.lazyval.integration.shared.Isbn;
import de.qualityminds.lazyval.integration.shared.Quantity;
import io.quarkus.hibernate.orm.panache.PanacheEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;

@Entity
@Table(name = "orders")
public class Order extends PanacheEntity {
    public Isbn isbn;
    // Quantity tests an boxed primitive from shared
    public Quantity quantity;
    public EMail email;

    public Order(Isbn isbn, Quantity quantity, EMail email) {
        this.isbn = isbn;
        this.quantity = quantity;
        this.email = email;
    }

    protected Order() {}
}

File: integration-test/kotlin/spring-app/src/main/kotlin/de/qualityminds/lazyval/integration/Order.kt
package de.qualityminds.lazyval.integration

import de.qualityminds.lazyval.integration.shared.Isbn
import de.qualityminds.lazyval.integration.shared.Quantity
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.springframework.data.jpa.repository.JpaRepository

@Table(name = "orders")
@Entity
class Order (
    var isbn: Isbn,
    var quantity: Quantity,
    var email: EMail,
){
    @Id
    @GeneratedValue
    var id: Long? = null
}

interface OrderRepository : JpaRepository<Order, Long>